aldryn_client/client.py

```diff
@@ -513,16 +518,99 @@ def workspace_init_virtualenv(self, sitename, path):
         if not os.path.exists(dev_path):
             os.mkdir(dev_path)
 
-        import subprocess
-        subprocess.call(['virtualenv', virtualenv_path])
+        # fix requirements
+        with open(requirements_path, 'r') as f:
+            lines = f.readlines()
 
-        subprocess.call([pip_path, 'install', '-r', requirements_path])
+        excluded = ['mercurial', 'bzr', 'gyp', 'pygobject', 'pygpgme', 'cffi']
+
+        with open(requirements_path, 'w') as f:
+            for line in lines:
+                if not any(re.match(pat, line) for pat in excluded):
+                    f.write(line)
+
+        subprocess.call(['virtualenv', virtualenv_path])
+        subprocess.call([pip_path, 'install', 'pyOpenSSL==0.15.1', 'ndg-httpsclient==0.3.3', 'pyasn1==0.1.7', 'cryptography==0.8.2'])
+        subprocess.call([pip_path, 'install', '--allow-all-external', '--allow-unverified', 'lazr.restfulclient', '-r', requirements_path])
         with open(os.path.join(virtualenv_path, 'lib/python2.7/site-packages/aldrynsite.pth'), 'w+') as fobj:
             fobj.write(os.path.abspath('.site/'))
         with open(os.path.join(virtualenv_path, 'lib/python2.7/site-packages/aldrynsite_dev.pth'), 'w+') as fobj:
             fobj.write(os.path.abspath(dev_path))
         return True, sitename
 
+    def load_db(self, sitename, path):
+        print 'loading db'
+        site_path = os.path.join(path, '.site')
+        tmp_path = os.path.join(path, '.tmp')
+        if not os.path.exists(tmp_path):
+            os.mkdir(tmp_path)
+        try:
+            response = self.session.get(
+                '/api/v1/workspace/%s/download/db/' % sitename, stream=True,
+                headers={'accept': 'application/x-tar-gz'})
+        except (requests.exceptions.ConnectionError,
+                requests.exceptions.Timeout):
+            return (False, Client.NETWORK_ERROR_MESSAGE)
+        if response.status_code != 200:
+            msgs = []
+            msgs.append("Unexpected HTTP Response %s" % response.status_code)
+            if response.status_code < 500:
+                msgs.append(response.content)
+            return (False, '\n'.join(msgs))
+
+        # unpack the downloaded tar.gz containing the database dump
+        print "extracting dump to .tmp"
+        from StringIO import StringIO
+        tar = tarfile.open(mode='r:gz', fileobj=StringIO(response.content))
+        tar.extractall(path=tmp_path)
+        tar.close()
+        print "finished extracting dump"
+
+        # Set up .env
+        database_name = 'aldryn_{}'.format(sitename)
```

Comment:
duh - fair enough. I was thinking about making them configurable, but must've forgotten that. Thanks for pointing it out.